### PR TITLE
Store joint transforms in EnvState structure

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This PR stores joint_transform map in the EnvState.

### CODE REVIEW

File: tesseract/tesseract_environment/src/kdl/kdl_state_solver.cpp [hunk 1]
```diff
@@ -218,21 +214,23 @@ void KDLStateSolver::calculateTransformsHelper(tesseract_common::TransformMap& t
     Eigen::Isometry3d local_frame, global_frame;
     KDLToEigen(current_frame, local_frame);
     global_frame = parent_frame * local_frame;
-    transforms[current_element.segment.getName()] = global_frame;
+    state.link_transforms[current_element.segment.getName()] = global_frame;
+    if (current_element.segment.getName() != scene_graph_->getRoot())
+      state.joint_transforms[current_element.segment.getJoint().getName()] = global_frame;
```
Comment (mpowelson) [inline tesseract/tesseract_environment/src/kdl/kdl_state_solver.cpp]: What does this do?

Comment (mpowelson) [reply tesseract/tesseract_environment/src/kdl/kdl_state_solver.cpp]: perhaps that should go in the detailed description of the docs?

Comment (marip8) [reply tesseract/tesseract_environment/src/kdl/kdl_state_solver.cpp]: Does this set both the link transform and joint transform to the same transformation (`global_frame`)?

Comment (Levi-Armstrong) [reply tesseract/tesseract_environment/src/kdl/kdl_state_solver.cpp]: Yea, so a child link of a joint has the same transform as the joint.

File: tesseract/tesseract_environment/include/tesseract_environment/core/types.h [hunk 2]
```diff
@@ -42,16 +42,28 @@ TESSERACT_COMMON_IGNORE_WARNINGS_POP
 
 namespace tesseract_environment
 {
-/** @brief This holds a state of the environment */
+/**
+ * @brief This holds a state of the environment
+ *
+ * It provides a way to look up the location of a link/joint in world coordinates system by link/joint name. It is
+ * possible to get the joint transform using the child link transfrom of the joint, but they are both provided for
```
Comment (mpowelson) [inline tesseract/tesseract_environment/include/tesseract_environment/core/types.h]: `transfrom` misspelled

**Comment by mpowelson** [comment]
Note that this is a breaking change. We should comment over on #263 to be friendly.

**Pull request merged**